packages/fiori/src/SideNavigationItem.ts

```diff
@@ -81,6 +81,28 @@ class SideNavigationItem extends SideNavigationSelectableItemBase {
 	@i18n("@ui5/webcomponents-fiori")
 	static i18nBundle: I18nBundle;
 
+	_initialChildDisabledStates: Map<SideNavigationItemBase, boolean> = new Map();
+
+	onBeforeRendering() {
+		this.items.forEach(item => {
+			if (!this._initialChildDisabledStates.has(item)) {
+				this._initialChildDisabledStates.set(item, item.disabled);
+			}
+		});
+
+		this._updateChildItemsDisabledState();
+	}
+
+	_updateChildItemsDisabledState() {
+		this.items.forEach(item => {
+			if (this.disabled) {
+				item.disabled = true;
+			} else {
+				item.disabled = this._initialChildDisabledStates.get(item)!;
+			}
+		});
+	}
```

Comment:
We should not modify items' disabled property based on another property. Attribute modification usually happens upon user interaction.
You should rather provide a getter in SideNavigationSubItem like this:
```ts
get effectiveDisabled() {
    return this.disabled || this._parentDisabled;
}
```
Use `effectiveDisabled` in the template of the sub items. Set `_parentDisabled` here.